cmd/agent/main.go

```diff
@@ -36,7 +36,8 @@ func main() {
 
 	tickerReport := time.NewTicker(time.Duration(cfg.ReportInterval) * time.Second)
 	defer tickerReport.Stop()
-
+	metricsCh := make(chan *[]model.MetricUnit)
+	errCh := make(chan error)
```

Comment:
Who will close it?

Reply:
So, please verify that resources won't leak.

Reply:
>>Who will close it? 
Me of course. 
Although I don't close these channels intentionally because they have to be alive while the agent exists and must die with it.